src/utils/crypto.ts

```diff
@@ -0,0 +1,148 @@
+import type { IEncryptionData } from '@/types';
+import { STORAGE_KEYS, PASSWORD_ENCRYPTION_ALGO, IS_EXTENSION } from '@/constants';
+import { WalletStorage } from '@/lib/WalletStorage';
+import { toRaw } from 'vue';
+
+/**
+ * If the application is running as an extension, the key should be extractable.
+ * So that we can save it in the session storage.
+ */
+const IS_EXTRACTABLE = !!IS_EXTENSION;
+
+// Utility Functions for Base64 Encoding and Decoding
+const encodeBase64 = (data: Uint8Array) => Buffer.from(data).toString('base64');
+const decodeBase64 = (data: string) => Buffer.from(data, 'base64');
+
+async function generateCryptoKey(password: string, salt: Uint8Array): Promise<CryptoKey> {
+  const subtleCrypto = globalThis.crypto.subtle;
+
+  // Encode password to ArrayBuffer
+  const enc = new TextEncoder();
+  const passwordBuffer = enc.encode(password);
+
+  // Derive the key using PBKDF2
+  const keyMaterial = await subtleCrypto.importKey(
+    'raw',
+    passwordBuffer,
+    { name: 'PBKDF2' },
+    false,
+    ['deriveKey'],
+  );
+
+  const key = await subtleCrypto.deriveKey(
+    {
+      name: 'PBKDF2',
+      salt,
+      iterations: 100000,
+      hash: 'SHA-256',
+    },
+    keyMaterial,
+    { name: PASSWORD_ENCRYPTION_ALGO, length: 256 },
+    IS_EXTRACTABLE,
+    ['encrypt', 'decrypt'],
+  );
+
+  return key;
+}
+
+export async function restoreEncryptionData(
+  password:string,
+  encryptedMnemonic: string,
+): Promise<IEncryptionData> {
+  const data = new Uint8Array(decodeBase64(encryptedMnemonic));
+  // Extract salt and iv from the data
+  const salt = data.slice(0, 16);
+  const iv = data.slice(16, 32);
+  const key = await generateCryptoKey(password, salt);
+  return { key, salt, iv };
+}
+
+export async function initializeEncryptionData(password: string): Promise<IEncryptionData> {
+  const salt = globalThis.crypto.getRandomValues(new Uint8Array(16));
+  const iv = globalThis.crypto.getRandomValues(new Uint8Array(16));
+  const key = await generateCryptoKey(password, salt);
+  return { key, salt, iv };
+}
+
+export async function encrypt(encryptionData: IEncryptionData, plaintext: string): Promise<string> {
+  const { key, salt, iv } = toRaw(encryptionData);
+  const subtleCrypto = globalThis.crypto.subtle;
+
+  // Encrypt the plaintext
+  const enc = new TextEncoder();
+  const encryptedBuffer = await subtleCrypto.encrypt(
+    {
+      name: PASSWORD_ENCRYPTION_ALGO,
+      iv,
+    },
+    key,
+    enc.encode(plaintext),
```

Comment:
As the encoder instance is used only once what do you think about using it directly here?

```suggestion
    new TextEncoder().encode(plaintext),
```